src/backend/src/services/projects.services.ts

```diff
@@ -811,4 +811,87 @@ export default class ProjectsService {
 
     return newMaterialType;
   }
+
+  /**
+   * Update a material
+   * @param name the name of the new material
+   * @param assembly the assembly of the new material
+   * @param status the Material status of the new material
+   * @param manufacturerName the manufacture name of the new material
+   * @param manufacturerPartNumber the manufacture part number of the new material
+   * @param pdmFileName the pdm file name of the new material
+   * @param quantity the quantity of the new material
+   * @param quantityUnit the quantity unit of the new material
+   * @param unitName the unit name of the new material
+   * @param price the price of the new material
+   * @param subtotal the subtotal of the new material
+   * @param linkUrl the link url of the new material
+   * @param notes the notes of the new material
+   */
+  static async editMaterial(
+    submitter: User,
+    wbsNumber: WbsNumber,
+    name: string,
+    status: Material_Status,
+    manufacturerName: string,
+    manufacturerPartNumber: string,
+    quantity: number,
+    unitName: string,
+    price: number,
+    subtotal: number,
+    linkUrl: string,
+    notes: string,
+    assembly?: Assembly,
+    pdmFileName?: string,
+    quantityUnit?: Unit
+  ): Promise<Material> {
+    const project = await prisma.project.findFirst({
+      where: {
+        wbsElement: {
+          carNumber: wbsNumber.carNumber,
+          projectNumber: wbsNumber.projectNumber,
+          workPackageNumber: wbsNumber.workPackageNumber
+        }
+      },
+      ...projectQueryArgs
+    });
+
+    if (!project) throw new NotFoundException('Project', wbsPipe(wbsNumber));
+    if (project.wbsElement.dateDeleted) throw new DeletedException('Project', project.projectId);
+
+    const perms = isLeadership(submitter.role) || isUserPartOfTeams(project.teams, submitter);
+
+    if (!perms) throw new AccessDeniedException('update material');
+
+    const updatedMaterial = await prisma.material.update({
+      where: { name },
+      data: {
+        status,
+        manufacturerName,
+        manufacturerPartNumber,
+        quantity,
+        unitName,
+        price,
+        subtotal,
+        linkUrl,
+        notes,
+        wbsElementId: project.wbsElementId,
+        // assembly: {
+        //   connect:
+        //     {
+        //       assembly
+        //     } || undefined
+        // },
+        pdmFileName: pdmFileName || undefined
+        // quantityUnit: {
+        //   connect:
```

Comment:
I believe you can set assemblyId and unitName directly and it should work the same, while avoiding this issue you're having